src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -157,7 +161,7 @@ private static void createStatus(@NonNull Run<?, ?> build, @NonNull TaskListener
             statusDescription = StringUtils.defaultIfBlank(buildDescription, "The build is in progress...");
             state = INPROGRESS_STATE;
         }
-        status = new BitbucketBuildStatus(hash, statusDescription, state, url, key, name);
+        status = new BitbucketBuildStatus(hash, statusDescription, state, url, DigestUtils.md5Hex(key), name);
```

Comment:
The added md5Hex call looked surprising but I now see you just moved it from the BitbucketBuildStatus constructor.